doc/rst/language/spec/tuples.rst

```diff
@@ -714,6 +715,286 @@ where a comma-separated list of components is valid.
+their default argument intent.
+
+If the tuple argument has the `in` intent, then it is a heavy tuple and all
+of its elements are captured by value, as though each element has the `in`
+intent.
+
+If a routine argument is a tuple with the default argument intent and a
+heavy tuple (such as a tuple variable) is passed to it, the heavy tuple
+will be silently converted into a light tuple.
+
+A conversion from light to heavy tuple also occurs when a light tuple
+(such as a tuple expression) is passed to a tuple argument with the `in`
+intent.
+
+   *Example (tuple-argument-behavior.chpl)*.
+
+Consider the following example:
+
+   .. code-block:: chapel
+
+      record R { var x: int; }
+
+      var modTup = (0, new R(0));
+
+      proc lightTupleArg(light) {
+
+        // Modify the module variable `modTup`.
+        modTup = (3, new R(6));
+
+        // Should print (0, (x = 6)).
+        writeln(light);
+
+        heavyTupleArg(light);
+
+        // Should still print (0, (x = 6)).
+        writeln(light);
+      }
+
+      proc heavyTupleArg(in heavy) {
+        heavy = (64, new R(128));
+      }
+
+      lightTupleArg(modTup);
+
+   .. BLOCK-test-chapeloutput
+
+      (0, (x = 6))
+      (0, (x = 6))
+
+The argument ``tup`` of the routine ``lightTupleArg`` is a light tuple due
+to the default argument intent. When the module variable ``modTup`` is
+passed to ``lightTupleArg``, its first element is copied while its second
+element is passed as though it were `const ref`.
+
+When the tuple argument ``tup`` is passed to ``heavyTupleArg``, a copy of it
+is made because the argument ``heavy`` is a heavy tuple due to the `in`
+intent.
+
+   *Example (tuple-argument-ref-intent.chpl)*.
+
+Tuple arguments with the `ref` intent are considered to be a special form
+of light tuple where every element is passed by reference.
+
+If a tuple argument has the `ref` intent, then actual arguments are
+restricted to heavy tuples (a tuple variable or a returned tuple). Each
+individual element behaves as though it was an argument with the `ref`
+intent.
+
+   .. code-block:: chapel
+
+      proc passTupleByRef(ref tup) {
+        tup = (64, 128);
+      }
+
+      var modTup = (0, 0);
+
+      passTupleByRef(modTup);
+
+      // Should print (64, 128).
+      writeln(modTup);
+
+   .. BLOCK-test-chapeloutput
+
+      (64, 128)
+
+When a tuple argument has the default argument intent, primitive types like
+integers are copied because their default intent is `in`. In the above
+example the intent of the formal argument ``tup`` is `ref`, and so the two
+integer elements of ``modTup`` are passed by reference. Any modifications
+made to the elements of ``tup`` will be visible when ``modTup`` is written
+to standard output.
+
+A tuple argument declared with `const` intent will work similarly to one
+with a default intent, except that all the elements of the tuple are
+considered to be `const` and cannot be modified.
```

Comment:
Maybe this section needs some sub-sub-sections? Anyway the two paragraphs above seem to belong better after "If the tuple argument has the default argument intent, then it is a light
tuple and some of its elements may be captured by `ref` depending on
their default argument intent."